numpy_challenge.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Tue Oct 26 17:28:51 2021
+
+@author: dmitriipodgalo
+"""
+
+import numpy as np
+
+
+if __name__ == "__main__":
+    np.array([1, 2, 3])
+    np.arange(1, 10, 0.5)
+    np.ones(10)
+
+
+def matrix_multiplication(m1, m2):
+    return np.array(m1) * np.array(m2)
+
+
+def multiplication_check(m):
+    m = np.array(m)
+
+    try:
+        for i in range(0, len(m) - 1):
+            m[i] * m[i + 1]
+    except ValueError:
+        return False
+    else:
+        return True
+
+
+def multiply_matrices(m):
+    m = np.array(m)
+    temp_matrix = np.array(m)[0]
+
+    try:
+        for i in range(1, len(m)):
+            temp_matrix *= m[i]
+    except ValueError:
+        return None
+    else:
+        return temp_matrix
```

Comment:
```suggestion
    if multiplication_check(m):
        for i in range(1, len(m)):
            temp_matrix *= m[i]
    else:
        raise ValueError
```
Так будет оптимальнее. В текущем варианте код `multiplication_check` и `multiply_matrices` практически идентичен